docs/architecture.md

```diff
@@ -0,0 +1,46 @@
+# source{d} lookout Architecture
+
+_You will find the project background, goals, constraints, solution details and other info in [**source{d} lookout DesignDoc**](https://docs.google.com/document/d/1pqz-_SHO5BsJE-aa8o_bAY3r5vR67amnWN8-qZc2UgY/edit#)_
+
+![lookout service sequence diagram](assets/lookout-seq-diagram.png)
```

Comment:
I don't like this diagram at all, to be honest. This one was in the docs at some point, then we removed it. Well, we moved it to a `.md` file that was not linked from anywhere, but should have been deleted.

I propose to replace it with this other one: [link](https://mermaidjs.github.io/mermaid-live-editor/#/edit/eyJjb2RlIjoic2VxdWVuY2VEaWFncmFtXG4gICAgcGFydGljaXBhbnQgR2l0SHViXG4gICAgcGFydGljaXBhbnQgbG9va291dFxuICAgIHBhcnRpY2lwYW50IEFuYWx5emVyXG4gICAgcGFydGljaXBhbnQgQmFiZWxmaXNoXG4gICAgbG9va291dC0-PkdpdEh1YjogUG9sbGluZ1xuICAgIEdpdEh1Yi0tPj5sb29rb3V0OiBQUi9QdXNoIGV2ZW50c1xuICAgIGxvb2tvdXQtPj5BbmFseXplcjogTm90aWZ5UmV2aWV3RXZlbnRcbiAgICBBbmFseXplci0-Pmxvb2tvdXQ6IEdldENoYW5nZXMvR2V0RmlsZXNcbiAgICBsb29rb3V0LT4-QmFiZWxmaXNoOiBwYXJzZUZpbGUgKGlmIFdhbnRVQVNUKVxuICAgIEJhYmVsZmlzaC0tPj5sb29rb3V0OiBVQVNUXG4gICAgbG9va291dC0tPj5BbmFseXplcjogQ2hhbmdlL0ZpbGVcbiAgICBBbmFseXplci0tPj5sb29rb3V0OiBDb21tZW50c1xuICAgIGxvb2tvdXQtPj5HaXRIdWI6IFBvc3QgY29tbWVudHMiLCJtZXJtYWlkIjp7InRoZW1lIjoiZGVmYXVsdCJ9fQ).

In my opinion, this one is overly-complicated. The goal of this diagram should be to provide a simple overview of the different components and how they interact with each other.
- What is that healthcheck loop exactly? And how is that relevant for an overview of the architecture?
- The registered repos/registered analyzers boxes add noise, without any added value
- DataServer is shown as a component. While this might be useful to describe the internals of the `lookoutd` code, from an architecture point of view this adds confusion. `Analyzer` could also be divided into `Analyzer gRPC Server`, `Analyzer` and `DataServer gRPC client`; but that would be just unnecesary noise. I think separating `lookout` and `DataServer` is in a similar category, too low-level for a diagram.

Reply:
About
> boxes are loop according to the label they have, meaning that something happens for every repo and every analyzer... because those events are not being processed by different lookout instances, but for the same one.
 
The boxes are correct, and I don't find them not misleading in any way. My feedback here is that visually it makes the diagram too busy, making it harder to understand things at a glance.

In this diagram I would say less is more. Saying that we poll GitHub for the registered repos is implied, just like we don't need to say that we'll use the configured credentials to poll GitHub. Or that we'll do that periodically in a loop; I think that's also imiplied.

Reply:
done by a178a5028745b566f4b013e29b668649805b754a